src/io/import_stl.cc

```diff
@@ -87,16 +88,15 @@ PolySet *import_stl(const std::string& filename, const Location& loc) {
   std::streampos file_size = f.tellg();
   f.seekg(80);
   if (f.good() && !f.eof()) {
-    uint32_t facenum = 0;
     f.read((char *)&facenum, sizeof(uint32_t));
 #if BOOST_ENDIAN_BIG_BYTE
     uint32_byte_swap(facenum);
 #endif
     if (file_size == static_cast<std::streamoff>(80ul + 4ul + 50ul * facenum)) {
       binary = true;
-      p->reserve(facenum);
     }
   }
+  PolySetBuilder builder(0,facenum);
```

Comment:
One bug: This `facenum` is only valid when binary is true.
This causes the `std::bad_alloc` because `facenum` is nonsense and too large.

Reply:
This is exactly the bug causing the CI build to fail. You can fix by changing it to
```c++
if (!binary) facenum = 0;
PolySetBuilder builder(0,facenum);
```